compose/service.py

```diff
@@ -430,12 +430,16 @@ def start_container(self, container):
         return container
 
     def connect_container_to_networks(self, container):
-        one_off = (container.labels.get(LABEL_ONE_OFF) == "True")
+        connected_networks = container.get('NetworkSettings.Networks')
 
         for network in self.networks:
+            if network in connected_networks:
+                self.client.disconnect_container_from_network(
+                    container.id, network)
```

Comment:
Yes, we need to set a `NetworkMode` or it'll default to `bridge` (and we'll just have to disconnect from that). I wondered about setting it to `none`, but it's then impossible to connect it to any networks.